features/net/FEATURE_IPV4/lwip-interface/lwipopts.h

```diff
@@ -66,6 +66,7 @@
 
 #define LWIP_DHCP                   1
 #define LWIP_DNS                    1
+#define LWIP_SOCKET                 0
```

Comment:
What does this modify? Seems unrelated to this patch or at least modifies behavior which I cannot tell.